torch/jit/frontend.py

```diff
@@ -346,17 +355,25 @@ def build_param_list(ctx, py_args, self_name):
             if arg is not None:
                 ctx_range = build_expr(ctx, arg).range()
                 raise NotSupportedError(ctx_range, _vararg_kwarg_err)
-    result = [build_param(ctx, arg, self_name, False) for arg in py_args.args]
-    result += [build_param(ctx, arg, self_name, True) for arg in py_args.kwonlyargs]
+
+    # List of Tuple of args and type as inferred by profile directed typing
+    arg_and_types = [(arg, next(iter(pdt_arg_types[arg.arg])) if pdt_arg_types[arg.arg] else None) for arg in py_args.args]
+    arg_and_types_kwonlyargs = [(arg, next(iter(pdt_arg_types[arg.arg])) if pdt_arg_types[arg.arg] else None)
+                                for arg in py_args.kwonlyargs]
+
+    result = [build_param(ctx, arg, self_name, kwarg_only=False, pdt_arg_type=_type) for arg, _type in arg_and_types]
```

Comment:
```suggestion
    result = [build_param(ctx, arg, self_name, kwarg_only=False, pdt_arg_type=type) for arg, type in arg_and_types]
```

Reply:
`type` is a builtin function so did not want to use that. Hence, used `_type`. But now changed to `arg_type`